security-analysis/security-analysis-api/src/main/java/com/powsybl/security/preprocessor/SecurityAnalysisPreprocessors.java

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.security.preprocessor;
+
+import com.google.common.io.ByteSource;
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.commons.config.PlatformConfig;
+import com.powsybl.commons.util.ServiceLoaderCache;
+import com.powsybl.contingency.ContingenciesProvider;
+import com.powsybl.contingency.ContingenciesProviderFactory;
+import com.powsybl.security.SecurityAnalysisConfig;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ *
+ * Provides instances of {@link SecurityAnalysisPreprocessor}s or factories.
+ *
+ * @author Sylvain Leclerc <sylvain.leclerc at rte-france.com>
+ */
+public final class SecurityAnalysisPreprocessors {
+
+    private SecurityAnalysisPreprocessors() {
+    }
+
+    /**
+     * Look for a factory with specified name, among the factories loaded as services.
+     * Throws if no matching factory is found.
+     *
+     * @param name the name identifying the requested factory.
+     * @return the factory corresponding to the specified name.
+     */
+    public static SecurityAnalysisPreprocessorFactory factoryForName(String name) {
+        List<SecurityAnalysisPreprocessorFactory> factories = new ServiceLoaderCache<>(SecurityAnalysisPreprocessorFactory.class).getServices();
+        return factories.stream()
+                .filter(f -> f.getName().equals(name))
+                .findFirst()
+                .orElseThrow(() -> new PowsyblException("Could not find any security analysis preprocessor for name " + name));
+    }
+
+    /**
+     * The default preprocessor factory, based on default platform {@link com.powsybl.commons.config.PlatformConfig config}.
+     *
+     * @return The preprocessor factory as defined in default platform config.
+     */
+    public static Optional<SecurityAnalysisPreprocessorFactory> defaultConfiguredFactory() {
```

Comment:
I think the name is misleading it is not the "default" configured factory, it is the factory in the configuration file (it can be the user's) not a default one in the framework/implementation. I get that it is to match `PlatformConfig.defaultConfig()` but then again, this method's name is a bit misleading as well. Maybe just call it `configuredFactory()` without parameter?